docs/pages/zkvm/sdk-benchmarking.mdx

```diff
@@ -0,0 +1,210 @@
+
+
+The host program `src/main.rs` is:
+
+```rust
+use nexus_sdk::{
+    compile::CompileOpts,
+    nova::seq::{Generate, Nova, PP},
+    Local, Prover, Verifiable,
+};
+
+const PACKAGE: &str = "guest";
+
+fn generate_pp_and_compile() -> (PP, Nova<Local>) {
+    let pp: PP = PP::generate().expect("failed to generate parameters");
+
+    let mut opts = CompileOpts::new(PACKAGE);
+    opts.set_memlimit(8); // use an 8mb memory
+
+    let prover: Nova<Local> = Nova::compile(&opts).expect("failed to compile guest program");
+
+    (pp, prover)
+}
+
+fn prove_execution(pp: &PP, prover: Nova<Local>) -> nexus_sdk::nova::seq::Proof {
+    prover.prove(pp).expect("failed to prove program");
+}
+
+fn verify_execution(pp: &PP, proof: &nexus_sdk::nova::seq::Proof) {
+    proof.verify(pp).expect("failed to verify proof");
+}
+
+fn main() {
+    use std::time::Instant;
+
+    let start = Instant::now();
+    let (pp, prover) = generate_pp_and_compile();
+    let duration = start.elapsed();
+    println!(
+        "Time taken to generate PP and compile: {:.2} seconds",
+        duration.as_secs_f64()
+    );
+
+    let start = Instant::now();
+    let proof = prove_execution(&pp, prover);
+    let duration = start.elapsed();
+    println!(
+        "Time taken to prove execution: {:.2} seconds",
+        duration.as_secs_f64()
+    );
+
+    let start = Instant::now();
+    verify_execution(&pp, &proof);
+    let duration = start.elapsed();
+    println!(
+        "Time taken to verify execution: {:.2} seconds",
+        duration.as_secs_f64()
+    );
+}
+```
+
+The guest program `src/guest/src/main.rs` is:
+
+```rust
+#![no_std]
+#![no_main]
+
+#[nexus_rt::profile]
+fn fibonacci(n: u32) -> u32 {
+    fib(n)
+}
+
+fn fib(n: u32) -> u32 {
+    match n {
+        0 => 0,
+        1 => 1,
+        _ => fib(n - 1) + fib(n - 2),
+    }
+}
+
+
+
+#[nexus_rt::main]
+fn main() {
+    let n = 13;
+    assert_eq!(233, fibonacci(n));
+}
+```
+
+`#[nexus_rt::profile]` macro is used to profile a function in the guest program.
+
+### Build flags
+
+Rust can automatically optimize the code for the current CPU architecture.
+In this example, we use the `-C target-cpu=native` flag to build `cargo-nexus` for the native CPU to optimize performance.
+If you want to set a specific CPUs, 
+you can obtain the list of target CPUs can be obtained from this command `rustc --print target-cpus`.
+
+```rust 
+RUSTFLAGS='-C target-cpu=native' cargo build --release --package cargo-nexus --bin cargo-nexus
+```
+
+We will use `cargo-nexus` to build the guest program in this example.
+
+### Fibonacci benchmark
+
+
+The results below are obtained in 2024-08 on a MacBook Air M1 with 16 Gb of RAM.
+
+#### Profile guest RISC-V Cycles count
+
+Run command at `src/guest/` directory: `cargo-nexus nexus run 2>/dev/null`
```

Comment:
grammar: "Run command at `src/guest/` directory:" -> "In the `src/guest/` directory, run the command:"

Reply:
Also, shouldn't this just be `cargo nexus ...`, here and below?

Reply:
grammar: updated in 915ef4c . 

The `cargo nexus run` is to get the trace cycle here.

For proving and verifying, I follow the SDK, at the `host`, a `cargo run -r` at root directory is enough. 
Otherwise it will be `cargo nexus prove` inside `/src/guest/` directory.  I prefer the SDK approach, cleaner.

Reply:
The complaint is solely about using `cargo-nexus`, as opposed to `cargo nexus ...`, which is a cleaner UI for the same functionality.

Reply:
I see. Updated in efdec84 . 
Also I removed the `#Build Flags` section, assuming user has followed the `SDK_quickstart.mdx`.